libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistorySetRouteMapper.kt

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.navigation.core.replay.history
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.navigation.core.history.MapboxHistoryReader
+import com.mapbox.navigation.core.history.model.HistoryEventSetRoute
+
+/**
+ * This class is used to replay the set route events from history files.
+ *
+ * When reading history from the [MapboxHistoryReader], there will be [HistoryEventSetRoute]
+ * included. To convert [HistoryEventSetRoute] into [ReplayEventBase], add an
+ * instance of this class to the [ReplayHistoryMapper.Builder.setRouteMapper].
+ */
+class ReplayHistorySetRouteMapper(
+    val accessToken: String
+) : ReplayHistoryEventMapper<HistoryEventSetRoute> {
+    override fun map(event: HistoryEventSetRoute): ReplayEventBase {
+        return ReplaySetRoute(
+            eventTimestamp = event.eventTimestamp,
+            route = mapDirectionsRoute(event.directionsRoute)
+        )
+    }
+
+    private fun mapDirectionsRoute(routeResponse: String?): DirectionsRoute? {
+        return if (routeResponse.isNullOrEmpty()) {
+            null
+        } else {
+            return DirectionsRoute.fromJson(routeResponse, accessToken)
+        }
```

Comment:
Minor thing. At this point is either `null` or a non-empty, no `{}` response.

```suggestion
        return routeResponse?.let {
            DirectionsRoute.fromJson(routeResponse, accessToken)
        } ?: return null
```

Reply:
Yeah you're right, the `HistoryEventMapper` handles these now. thanks! can also remove the `?: return null`